### PR TITLE
Unittest for inventory/v1

### PR DESCRIPTION
**Pull request by hugonicodemos**

### CODE REVIEW

**Comment by yjian118** [comment]
Please rebase and resolve merge conflicts

File: starlingx/inventory/v1/storagetiers/requests.go [hunk 1]
```diff
@@ -45,8 +45,8 @@ func (opts ListOpts) ToStorageTierListQuery() (string, error) {
 // List returns a Pager which allows you to iterate over a collection of
 // StorageTiers. It accepts a ListOpts struct, which allows you to filter
 // and sort the returned collection for greater efficiency.
-func List(c *gophercloud.ServiceClient, clusterID string, opts ListOptsBuilder) pagination.Pager {
-	url := listURL(c, clusterID)
+func List(c *gophercloud.ServiceClient, cluster string, opts ListOptsBuilder) pagination.Pager {
```
Comment (yjian118) [inline starlingx/inventory/v1/storagetiers/requests.go:48]: Would appreciate idea from @ttakenak

**Review by yjian118 — APPROVED** [review]
lgtm

File: starlingx/inventory/v1/storagetiers/requests.go (hunk 1 above)
Comment (ttakenak) [reply starlingx/inventory/v1/storagetiers/requests.go:48]: Isn't it just update the parameter name?

File: starlingx/inventory/v1/interfaceDataNetworks/results.go [hunk 2]
```diff
@@ -50,11 +50,11 @@ type InterfaceDataNetwork struct {
 
 	// DataNetworkUUID defines the system assigned unique UUID of the associated
 	// data network.
-	DataNetworkUUID string `json:"network_uuid"`
+	DataNetworkUUID string `json:"datanetwork_uuid"`
```
Comment (ttakenak) [inline starlingx/inventory/v1/interfaceDataNetworks/results.go:53]: Is this ok to change json name?

File: starlingx/inventory/v1/interfaceDataNetworks/requests.go [hunk 3]
```diff
@@ -101,3 +101,9 @@ func ListInterfaceDataNetworks(c *gophercloud.ServiceClient, hostid string) ([]I
 
 	return objs, err
 }
+
+// Get retrieves a specific interfaceDataNetwork based on its unique ID.
+func Get(c *gophercloud.ServiceClient, id string) (r GetResult) {
```
Comment (ttakenak) [inline starlingx/inventory/v1/interfaceDataNetworks/requests.go:106]: I think it is ok but do we need to add this method?

**Comment by ttakenak** [comment]
Is this PR for adding test? Or include some fix? I commented two above but I see adding method in several requests.go (mainly GET). I think it is harmless but do we need to add them?

File: starlingx/inventory/v1/interfaceDataNetworks/results.go (hunk 2 above)
Comment (yjian118) [reply starlingx/inventory/v1/interfaceDataNetworks/results.go:53]: The "datanetwork_uuid" should be correct according to: https://docs.starlingx.io/api-ref/config/api-ref-sysinv-v1-config.html#interface-data-networks

File: starlingx/inventory/v1/osds/testing/requests_test.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	JournalSize := 2048
+	actual, err := osds.Update(client.ServiceClient(),
+		"dfa582bd-8f1d-4e8c-9c8a-1167243d592a",
+		osds.OSDOpts{JournalLocation: &JournalLocation, JournalSize: &JournalSize}).Extract()
+	if err != nil {
+		t.Fatalf("Unexpected Update error: %v", err)
+	}
+	th.CheckDeepEquals(t, OSDDerp, *actual)
+}
+*/
```
Comment (yjian118) [inline starlingx/inventory/v1/osds/testing/requests_test.go]: May need a new line to correct the format problem. Or better to uncomment this section

File: starlingx/inventory/v1/partitions/testing/requests_test.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	size_mib := 400
+	actual, err := partitions.Update(client.ServiceClient(),
+		"09aa5a82-aa89-4d86-a9d1-7410167d510b",
+		partitions.DiskPartitionOpts{Size: size_mib}).Extract()
+	if err != nil {
+		t.Fatalf("Unexpected Update error: %v", err)
+	}
+	th.CheckDeepEquals(t, DiskPartitionDerp, *actual)
+}
+*/
```
Comment (yjian118) [inline starlingx/inventory/v1/partitions/testing/requests_test.go]: same here

**Comment by yjian118** [comment]
> Is this PR for adding test? Or include some fix? I commented two above but I see adding method in several requests.go (mainly GET). I think it is harmless but do we need to add them?

The main motivation is to merge the PR #3 and #4 to increase the test coverage, fix the unit tests and merge conflicts. I'm okay to introduce the GET methods though not used, we can correct that if the get methods have problems. 

@hugonicodemos please correct the format issue and ensure "go vet ./..." and "go test ./..." in starlingx directory

**Comment by ttakenak** [comment]
ok, @yjian118 Thank you for your comment.

**Comment by hugonicodemos** [comment]
> > Is this PR for adding test? Or include some fix? I commented two above but I see adding method in several requests.go (mainly GET). I think it is harmless but do we need to add them?
> 
> The main motivation is to merge the PR #3 and #4 to increase the test coverage, fix the unit tests and merge conflicts. I'm okay to introduce the GET methods though not used, we can correct that if the get methods have problems.
> 
> @hugonicodemos please correct the format issue and ensure "go vet ./..." and "go test ./..." in starlingx directory

Done

**Review by yjian118 — APPROVED** [review]

**Review by ttakenak — APPROVED** [review]

**Pull request merged**